pkg/registry/apis/provisioning/jobs/pullrequest/worker.go

```diff
@@ -24,25 +24,29 @@ type PullRequestRepo interface {
 }
 
 type PullRequestWorker struct {
-	parsers   resources.ParserFactory
-	previewer Previewer
+	parsers        resources.ParserFactory
+	renderer       ScreenshotRenderer
+	urlProvider    func(namespace string) string
+	commentBuilder *commentBuilder
 }
 
 func NewPullRequestWorker(
 	parsers resources.ParserFactory,
-	previewer Previewer,
+	renderer ScreenshotRenderer,
```

Comment:
See my comments below about the Commenter and ChangeEvaluator. 

```go
NewPullRequestWorker(evaluator ChangeEvaluator, commenter Commenter) ... 
```

Reply:
I am not yet seeing how the clean things up -- so I'll save it for later.  IIUC that means we have to make two public interfaces that get initialized in the registry passing in all the same dependencies.